compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -969,7 +969,9 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                 self.check_edge(location, *target, EdgeKind::Normal);
                 self.check_unwind_edge(location, *unwind);
             }
-            TerminatorKind::Call { func, args, destination, target, unwind, .. } => {
+
+            TerminatorKind::Call { func, args, .. }
+            | TerminatorKind::TailCall { func, args, .. } => {
```

Comment:
should we do some extra validation here that checks that the tail called function has the same signature as the current function?

Reply:
This would duplicate the checks with the ones that are run on thir (again, yet to make a PR...), but I guess if we think that MIR opts might screw things up accidentally adding a check here would de appropriate.

Reply:
That, and custom_mir 😆 it doesn't have to be in this PR, so leaving a FIXME is fine by me

Reply:
Ended up adding a fixme, as it's a bit tricky to implement 3937e47